### ISSUE TITLE
segfault from running ai-benchmark on MBP 16"

### ISSUE THREAD

**Issue by Zacharias030** [apple/tensorflow_macos#95]
Just running the standard `ai-benchmark` causes a segfault on the first training task.

reproduce after `$ pip install ai-benchmark`
```
(tf_mac) ➜  tensorflow_macos python
Python 3.8.0 (default, Nov  6 2019, 15:49:01)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from ai_benchmark import AIBenchmark
>>> benchmark = AIBenchmark()

>>   AI-Benchmark-v.0.1.2
>>   Let the AI Games begin..

>>>
>>>
>>>
>>> results = benchmark.run()
*  TF Version: 2.4.0-rc0
*  Platform: macOS-10.16-x86_64-i386-64bit
*  CPU: N/A
*  CPU RAM: 16 GB

The benchmark is running...
The tests might take up to 20 minutes
Please don't interrupt the script

1/19. MobileNet-V2

1.1 - inference | batch=50, size=224x224: 804 ± 35 ms
[1]    4458 segmentation fault  python
(tf_mac) ➜  tensorflow_macos ls
```

**Comment by anna-tikhonova** [comment]
Thank you very much for reporting this. We will investigate and report back.

**Comment by ma010** [comment]
I got a similar error (reported here https://github.com/apple/tensorflow_macos/issues/25#issuecomment-752263711) when running a script from that same thread. The error would occur in any of these scenarios:
1.
```
from tensorflow.python.framework.ops import disable_eager_execution
disable_eager_execution()
```
2.
```
from tensorflow.python.compiler.mlcompute import mlcompute
mlcompute.set_mlc_device(device_name='any')
```
3. Using dropout such as:
```
tf.keras.layers.Dropout(0.25)
```

Script used is here:
```
import tensorflow.compat.v2 as tf
import tensorflow_datasets as tfds

tf.enable_v2_behavior()

from tensorflow.python.framework.ops import disable_eager_execution
#disable_eager_execution()

from tensorflow.python.compiler.mlcompute import mlcompute
mlcompute.set_mlc_device(device_name='any')


(ds_train, ds_test), ds_info = tfds.load(
    'mnist',
    split=['train', 'test'],
    shuffle_files=True,
    as_supervised=True,
    with_info=True,
)

def normalize_img(image, label):
    """Normalizes images: `uint8` -> `float32`."""
    return tf.cast(image, tf.float32) / 255., label

batch_size = 128

ds_train = ds_train.map(
    normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)
ds_train = ds_train.cache()
ds_train = ds_train.shuffle(ds_info.splits['train'].num_examples)
ds_train = ds_train.batch(batch_size)
ds_train = ds_train.prefetch(tf.data.experimental.AUTOTUNE)


ds_test = ds_test.map(
    normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)
ds_test = ds_test.batch(batch_size)
ds_test = ds_test.cache()
ds_test = ds_test.prefetch(tf.data.experimental.AUTOTUNE)


model = tf.keras.models.Sequential([
    tf.keras.layers.Conv2D(32, kernel_size=(3, 3),
                           activation='relu'),
    tf.keras.layers.Conv2D(64, kernel_size=(3, 3),
                           activation='relu'),
    tf.keras.layers.MaxPooling2D(pool_size=(2, 2)),
    # tf.keras.layers.Dropout(0.25),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(128, activation='relu'),
    # tf.keras.layers.Dropout(0.5),
    tf.keras.layers.Dense(10, activation='softmax')
])
model.compile(
    loss='sparse_categorical_crossentropy',
    optimizer=tf.keras.optimizers.Adam(0.001),
    metrics=['accuracy'],
)

model.fit(
    ds_train,
    epochs=12,
    validation_data=ds_test,
)
```
